File: qb-engineer-server/qb-engineer.core/Models/UpdateAssetRequestModel.cs
using QBEngineer.Core.Enums;

namespace QBEngineer.Core.Models;

public record UpdateAssetRequestModel(
    string? Name,
    AssetType? AssetType,
    string? Location,
    string? Manufacturer,
    string? Model,
    string? SerialNumber,
    AssetStatus? Status,
    decimal? CurrentHours,
    string? Notes,
    bool? IsCustomerOwned,
    int? CavityCount,
    int? ToolLifeExpectancy,
    int? CurrentShotCount,
    int? SourceJobId,
    int? SourcePartId);

File: qb-engineer-ui/src/app/features/assets/models/update-asset-request.model.ts
import { AssetType } from './asset-type.type';
import { AssetStatus } from './asset-status.type';

export interface UpdateAssetRequest {
  name?: string;
  assetType?: AssetType;
  location?: string;
  manufacturer?: string;
  model?: string;
  serialNumber?: string;
  status?: AssetStatus;
  currentHours?: number;
  notes?: string;
  isCustomerOwned?: boolean;
  cavityCount?: number;
  toolLifeExpectancy?: number;
}
